src/lib.rs

```diff
@@ -618,13 +626,38 @@ impl GenerateTasksService for GenerateTasksServiceImpl {
             for task in &build_variant.tasks {
                 if task.name == BURN_IN_TAGS {
                     if self.gen_burn_in {
-                        burn_in_tag_build_variant_names.extend(
-                            self.evg_config_utils
-                                .lookup_and_split_by_whitespace_build_variant_expansion(
-                                    BURN_IN_TAG_BUILD_VARIANTS,
-                                    build_variant,
-                                ),
-                        );
+                        let burn_in_tag_build_variants = self
+                            .evg_config_utils
+                            .lookup_and_split_by_whitespace_build_variant_expansion(
+                                BURN_IN_TAG_BUILD_VARIANTS,
+                                build_variant,
+                            );
+                        let compile_distro = self
+                            .evg_config_utils
+                            .lookup_build_variant_expansion(
+                                BURN_IN_TAG_COMPILE_DISTRO,
+                                build_variant,
+                            )
+                            .unwrap_or_else(|| DEFAULT_BURN_IN_COMPILE_TASK_DISTRO.to_string());
+                        let compile_task_group_name = self
+                            .evg_config_utils
+                            .lookup_build_variant_expansion(
+                                BURN_IN_TAG_COMPILE_TASK_GROUP_NAME,
+                                build_variant,
+                            )
+                            .unwrap_or_else(|| DEFAULT_BURN_IN_COMPILE_TASK_GROUP_NAME.to_string());
+                        for variant in burn_in_tag_build_variants {
+                            let bv_info = burn_in_tag_build_variant_info.entry(variant).or_insert(
+                                BurnInTagBuildVariantInfo {
+                                    compile_distros: HashSet::from([compile_distro.to_string()]),
+                                    compile_task_group_name: compile_task_group_name.clone(),
+                                },
+                            );
+                            bv_info.compile_distros.insert(compile_distro.to_string());
```

Comment:
Do we want to apply the same logic here and add only non-default distros here as below?

Reply:
I don't think we can avoid adding the default here like below. We'll always need at least one distro to send to Evergreen and if that distro isn't set in the Evergreen yaml, we'll need to add the default here.

What do you think?

Reply:
I was thinking that at least one distro will be already added before this line. Does it look right?

Reply:
Yes, there should at least be one distro by the time this line hits. That distro could be the default distro though. I could add some logic here that, if the only distro is the default one, we replace the default with the non default distro. I think that might not be the best case though as theoretically someone could define the default distro as part of the evergreen yaml and we would be removing it.

The only time this should be a problem is if multiple different variants define the same burn_in_tag_build_variants. Which honestly seems like a bit of a strange case in my mind.
https://github.com/10gen/mongo/blob/master/etc/evergreen.yml#L1257
https://github.com/10gen/mongo/blob/master/etc/evergreen.yml#L1594
Here are the two places where we're currently defining the same burn_in variants. I think it would be fine to remove one?

Reply:
I think we should allow to have the same `burn_in_tag_buildvariants` defined in different places.

My understanding of burn_in_tags is that the user is just choosing to run all the tasks on some buildvariant and it will seamlessly generate burn_in_tags buildvariants. I think the majority of users should not be bothered selecting specifically `burn_in_tags_gen` task. So with the current evergreen configuration it will happen on `enterprise-rhel-80-64-bit-dynamic-required` and `enterprise-rhel-80-64-bit-dynamic-classic-engine` buildvariants and those generated buildvariants are happen to be the same.

Here is what I suggest:
1. Let's remove default values. We will explicitly define them for existing tasks and it will be a good example of how to configure some new tasks. 
2. Enforce consistent configuration. If there are several definitions of `burn_in_tag_buildvariants`, error out when the values of `burn_in_tag_compile_distro` and `burn_in_tag_compile_task_group_name` are different across those several definitions.

Let me know what you think.

Reply:
That sounds like a good path forward to me